src/manifest.rs

```diff
@@ -25,21 +27,58 @@ impl Manifest {
     /// Create an initial manifest for a new clone. The returned
     /// manifest will set `initialized=false` to allow for additional
     /// initialization (such as copying wals).
-    pub(crate) fn cloned(parent_db: ParentDb, parent_manifest: &Manifest) -> Self {
-        let clone_core = parent_manifest.core.init_clone_db();
+    pub(crate) fn cloned(
+        parent_path: String,
+        source_checkpoint_id: Uuid,
+        final_checkpoint_id: Uuid,
+        parent_manifest: &Manifest,
+    ) -> Self {
+        let mut parent_external_sst_ids = HashSet::new();
+        let mut clone_external_dbs = vec![];
+
+        for parent_external_db in &parent_manifest.external_dbs {
+            parent_external_db.sst_ids.iter().for_each(|id| {
+                parent_external_sst_ids.insert(*id);
+            });
+            clone_external_dbs.push(ExternalDb {
+                path: parent_external_db.path.clone(),
+                source_checkpoint_id: parent_external_db.source_checkpoint_id,
+                final_checkpoint_id,
+                sst_ids: parent_external_db.sst_ids.clone(),
+            });
+        }
+
+        let parent_owned_sst_ids = parent_manifest
+            .core
+            .compacted
+            .iter()
+            .flat_map(|sr| sr.ssts.iter().map(|s| s.id))
+            .chain(parent_manifest.core.l0.iter().map(|s| s.id))
+            .filter(|id| !parent_external_sst_ids.contains(id))
+            .collect();
+
+        clone_external_dbs.push(ExternalDb {
+            path: parent_path,
+            source_checkpoint_id,
+            final_checkpoint_id,
+            sst_ids: parent_owned_sst_ids,
+        });
+
         Self {
-            parent: Some(parent_db),
-            core: clone_core,
+            external_dbs: clone_external_dbs,
+            core: parent_manifest.core.init_clone_db(),
```

Comment:
Just want to confirm my understanding: the SRs are maintained here because the parent core has the SRs for its SSTs (and transitively, all previously cloned manifests, as well).

Reply:
yes, we can just copy SRs over from the parent